### PR TITLE
Show quick doc for implicit lambda parameter 'it'

### PR DESCRIPTION
**Pull request by cypressious**
Fixes #KT-9271

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/KotlinQuickDocumentationProvider.kt [hunk 1]
```diff
@@ -114,15 +116,28 @@ class KotlinQuickDocumentationProvider : AbstractDocumentationProvider() {
...
                 return "No documentation available"
             }
 
+            return renderKotlin(context, declarationDescriptor, quickNavigation)
+        }
+
+        private fun renderKotlinImplicitLambdaParameter(element: KtReferenceExpression, quickNavigation: Boolean): String? {
+            val context = element.analyze(BodyResolveMode.PARTIAL)
+            val target = element.mainReference.resolveToDescriptors(context).singleOrNull() as? ValueParameterDescriptor? ?: return null
+            context.get(BindingContext.AUTO_CREATED_IT, target)
```
Comment (yole) [inline idea/src/org/jetbrains/kotlin/idea/KotlinQuickDocumentationProvider.kt]: This line doesn't do anything

File: idea/src/org/jetbrains/kotlin/idea/KotlinQuickDocumentationProvider.kt [hunk 2]
```diff
@@ -114,15 +116,28 @@ class KotlinQuickDocumentationProvider : AbstractDocumentationProvider() {
...
+            return renderKotlin(context, target, quickNavigation)
+        }
+
+        private fun renderKotlin(context: BindingContext, declarationDescriptor: DeclarationDescriptor, quickNavigation: Boolean): String {
+            @Suppress("NAME_SHADOWING")
+            var declarationDescriptor = declarationDescriptor
```
Comment (yole) [inline idea/src/org/jetbrains/kotlin/idea/KotlinQuickDocumentationProvider.kt]: This doesn't seem to be needed, either

**Comment by yole** [comment]
Looks good except for the noted issues.

**Comment by cypressious** [comment]
@yole fixed

**Pull request merged**